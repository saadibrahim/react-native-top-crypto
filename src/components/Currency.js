import React, { Component } from 'react';
import { View, Text, Image } from 'react-native';
import numeral from 'numeral';
import PropTypes from 'prop-types';
import ICONS from '../constants/icons';
import styles from './styles/CurrencyStyles';
import { currencyFormatter } from '../helpers/numbers';

export default class Currency extends Component {
  render() {
    const { currency, index } = this.props;
    const { symbol, name, quote } = currency;
    const { price, market_cap: marketCap, percent_change_24h: percentChange } = quote.USD;

    return (
      <View style={styles.wrapper}>
        <Text style={styles.position}>#{index + 1}</Text>
        <View style={styles.nameWrapper}>
          <Image style={styles.icon} source={ICONS[symbol.toLowerCase()]} />
          <Text style={styles.symbol}>{symbol}</Text>
          <Text style={styles.name}>{name}</Text>
        </View>
        <Text style={styles.price}>
          {`Price: ${currencyFormatter.format(price)} `}
          <Text style={percentChange < 0 ? styles.negativeChange : styles.positiveChange}>
            ({percentChange > 0 && '+'}
            {percentChange.toFixed(3)}%)
          </Text>
        </Text>
        <Text>Market Cap: {numeral(marketCap).format('($0.00a)')}</Text>
      </View>
    );
  }
}

Currency.propTypes = {
  currency: PropTypes.shape({
    symbol: PropTypes.string,
    name: PropTypes.string,
    quote: PropTypes.shape({
      USD: PropTypes.shape({
        price: PropTypes.number,
        market_cap: PropTypes.number,
        percent_change_24h: PropTypes.number,
      }),
    }),
  }).isRequired,
  index: PropTypes.number.isRequired,
};
