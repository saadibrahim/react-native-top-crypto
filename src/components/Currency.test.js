import React from 'react';
import renderer from 'react-test-renderer';

import Currency from './Currency';

const currency = {
  symbol: 'BTC',
  name: 'Bitcoin',
  quote: {
    USD: {
      price: 10000,
      market_cap: 1000000,
      percent_change_24h: -1.11,
    },
  },
};

test('renders the Currency using Snapshots', () => {
  const component = renderer.create(<Currency currency={currency} index={1} />);
  expect(component).toMatchSnapshot();
});
