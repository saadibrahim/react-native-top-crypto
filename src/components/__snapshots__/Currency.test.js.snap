// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Currency using Snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 5,
      "margin": 10,
      "marginBottom": 0,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#888",
        "fontStyle": "italic",
        "fontWeight": "bold",
        "position": "absolute",
        "right": 10,
        "top": 10,
      }
    }
  >
    #
    2
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/resources/icons/btc.png",
        }
      }
      style={
        Object {
          "height": 24,
          "marginRight": 7,
          "width": 24,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "600",
          "marginRight": 7,
        }
      }
    >
      BTC
    </Text>
    <Text
      style={
        Object {
          "color": "#888",
          "fontSize": 14,
        }
      }
    >
      Bitcoin
    </Text>
  </View>
  <Text
    style={
      Object {
        "marginVertical": 3,
      }
    }
  >
    Price: $10,000.00 
    <Text
      style={
        Object {
          "color": "#FF0000",
        }
      }
    >
      (
      -1.110
      %)
    </Text>
  </Text>
  <Text>
    Market Cap: 
    $1.00m
  </Text>
</View>
`;
